docs/network/calico-network.md

```diff
@@ -0,0 +1,196 @@
+    ```
+
+-  *Calico nodes*
+
+    ```
+    kube-system     calico-node-ct6c9   2/2     Running
+    ```
+
+> Note: When you join a node to the Kubernetes cluster, a new *Calico node* is initiated on the Kubernetes node.
+
+### Configure BGP peer with Ingress Citrix ADC
+
+After you have established the Calico network between the master and the other nodes in the cluster, whenever you deploy an application, Kubernetes uses an IP address from the IP address pool of Calico and assigns it to the service associated with the application.
+
+[Border Gateway Protocol (BGP)](https://en.wikipedia.org/wiki/Border_Gateway_Protocol) uses [autonomous system number (AS number)](https://en.wikipedia.org/wiki/Autonomous_system_(Internet)) to identify the remote nodes. The AS number is a special number assigned by IANA used primarily with BGP to identify a network under a single network administration that uses unique routing policy.
+
+#### Configure BGP on Kubernetes using Ingress Citrix ADC
+
+Using a YAML file, you can apply BGP configuration of a remote node using the `kubectl create` command. In the YAML file, you need to add the peer IP address and the AS number. The peer IP address is the Ingress Citrix ADC IP address and the AS number is the AS number that is used in the Ingress Citrix ADC.
+
+#### Obtain the AS Number of the cluster
+
+Using the Calico ctl, you can obtain the AS number used in the Kubernetes cluster by Calico BGP as shown in the following image:
+
+![Get AS number](../media/as-number.png)
+
+#### Configure global BGP peer
+
+Using the `calicoctl` utility you can peer Calico nodes with global BGP speakers. This kind of peers is called global peers.
+
+Create a YAML definition file called `bgp.yml` with the following definition:
+
+```yml
+apiVersion: projectcalico.org/v3  # This is the version of Calico
+kind: BGPPeer  # BGPPeer specifies that its Global peering.
+metadata:
+    name: bgppeer-global-3040  # The name of the configuration
+spec:
+    peerIP: 10.102.33.208  # IP address of the Ingress Citrix ADC
+    asNumber: 500  # AS number configured on the Ingress Citrix ADC
+```
+
+Deploy the definition file using the following command:
+
+    > kubectl create -f bgp.yml
+
+#### Add the BGP configurations on the Ingress Citrix ADC
+
+Perform the following:
+
+1.  Log on to the Citrix ADC command-line interface.
+
+1.  Enable the BGP feature using the following command:
+
+        > en feature bgp
+          Done
+
+1.  Type `vtysh` and press **Enter**.
+
+        > vtysh
+        ns#
+
+1.  Change to config terminal using the `conf t` command:
+
+        ns#conf t
+        Enter configuration commands, one per line. End with CNTL/Z.
+        ns(config)#
+
+1.  Add the BGP route with the AS number as 500 for demonstration purpose. You can use any number as AS number.
+
+        ns(config)#router bgp 500
+        ns(config-router)#
+
+1.  Add neighbors using the following command:
```

Comment:
As we are already using this setup I have a small addendum. Due to the fact that the ADC only allows route advertisement from known neighbors you'll have to add all nodes from the k8s cluster to the neighbor table. Otherwise pods running on nodes that are not in the neighbor table will not be known to the ADC and will result in partial-down or down service groups.

As there is no listen-range as there is on Cisco you will have to manually add new nodes to the list or simply add the whole possible range of IPs that could be used as node IPs, which is what we did.